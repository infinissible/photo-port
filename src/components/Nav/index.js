import React from 'react';

function Nav() {
  function categorySelected(name) {
    console.log(`${name} clicked`);
  }
  return (
    <header>
      <h2>
        <a href="/">
          <span role="img" arial-label="camera">
            ðŸ“¸
          </span>{' '}
          Oh Snap!
        </a>
      </h2>
      <nav>
        <ul className="flex-row">
          <li className="mx-2">
            <a href="#about">About me</a>
          </li>
          <li>
            <span>Contact</span>
          </li>
          {categories.map((category) => (
            <li className="mx-1" key={category.name}>
              <span
                onClick={() => {
                  categorySelected(category.name);
                }}
              >
                {category.name}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

const categories = [
  {
    name: 'Commercial',
    description:
      'Photos of grocery stores, food trucks, and other commercial projects',
  },
  {
    name: 'Portraits',
    description: 'Portraits of people in my life',
  },
  {
    name: 'Food',
    description: 'Delicious delicacies',
  },
  {
    name: 'Landscape',
    description: 'Fields, farmhouses, waterfalls, and the beauty of nature',
  },
];

export default Nav;
