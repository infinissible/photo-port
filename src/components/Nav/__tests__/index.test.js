import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
  // basline test
  it('renders', () => {
    render(<Nav />);
  });

  // snapshot test
  it('matches snaphot', () => {
    const { asFragment } = render(<Nav />);
    // assert value comparison
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('emoji is visible', () => {
  it('inserts emoji into the h2', () => {
    const { getByLabelText } = render(<Nav />);

    expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
  });
});

describe('links are visible', () => {
  it('inserts text into the links', () => {
    // Arrange
    const { getByTestId } = render(<Nav />);

    // Assert
    expect(getByTestId('link')).toHaveTextContent('Oh Snap!');
    expect(getByTestId('about')).toHaveTextContent('About me');
  });
});
