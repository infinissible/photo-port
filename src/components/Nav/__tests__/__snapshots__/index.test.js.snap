// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snaphot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        href="/"
      >
        <span
          arial-label="camera"
          role="img"
        >
          ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        href="/"
      >
        <span
          arial-label="camera"
          role="img"
        >
          ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
